### ISSUE TITLE
[BUG] wings.async_call_scheduler - ERROR - API call error.

### ISSUE THREAD

**Issue by carlolm** [CoinAlpha/hummingbot#209]
**Describe the bug**

After starting the bot, the error was logged.  It does not appear that this error has prevented the bot from continuing to run.

```
2019-05-10 22:32:09,650 - wings.async_call_scheduler - ERROR - API call error.
Traceback (most recent call last):
  File "/Users/carlol/Github/coinalpha/hummingbot/wings/async_call_scheduler.py", line 76, in _coro_scheduler
    fut.set_result(await coro)
asyncio.base_futures.InvalidStateError: invalid state
```

Full log:

```
2019-05-10 22:32:02,721 - wings.market.binance_market - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2019-05-10 22:32:03,009 - hummingbot.strategy.arbitrage.arbitrage - WARNING - Markets are not ready. No arbitrage trading is permitted.
2019-05-10 22:32:09,650 - wings.async_call_scheduler - ERROR - API call error.
Traceback (most recent call last):
  File "/Users/carlol/Github/coinalpha/hummingbot/wings/async_call_scheduler.py", line 76, in _coro_scheduler
    fut.set_result(await coro)
asyncio.base_futures.InvalidStateError: invalid state
2019-05-10 22:32:11,629 - wings.wallet.web3_wallet - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2019-05-10 22:32:13,339 - wings.market.ddex_market - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2019-05-10 22:32:13,617 - wings.wallet.web3_wallet_backend - INFO - Approval already exists for Dai Stablecoin v1.0 from wallet address <****WALLET****>.
2019-05-10 22:32:13,617 - wings.wallet.web3_wallet_backend - INFO - Approval already exists for Tether USD from wallet address <****WALLET****>.
2019-05-10 22:32:13,617 - wings.wallet.web3_wallet_backend - INFO - Approval already exists for Wrapped Ether from wallet address <****WALLET****>.

```

**Steps To Reproduce**

Started arbitrage strategy with binance and ddex.

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters: ...
2. Start the bot using this command: ...
...

**Screenshots**
n/a

**Release version**

dev/0.6.0 last commit: 0d23b5f3358440a674c6f3f6871d0697416f02cf May 10, 2019

**Attachments**

**Comment by carlolm** [comment]
Error also logged with running xemm.  Seems like network/api was not yet ready.

**Comment by louzhixian** [comment]
same issue here

**Comment by martinkou** [comment]
This is likely caused by the API call returning after the future object has been cancelled in another coroutine. The error log is likely harmless if that's the case.

It is still wrong to display spurious error logs. I'm going to look at this along with the logging review during this sprint.

**Comment by martinkou** [comment]
Fixed with #233.
